examples/recipes/kimi/pretrain_kimi_k2.py

```diff
@@ -0,0 +1,198 @@
+    4. Starts Megatron pretraining with the final merged configuration
+
+    Configuration merging preserves callable fields (like activation functions)
+    and handles type conversions automatically.
+
+    Examples of CLI usage:
+        # Use default config with custom learning rate
+        python pretrain_kimi_k2.py optimizer.lr=0.0003
+
+        # Custom config file with additional overrides
+        python pretrain_kimi_k2.py --config-file my_config.yaml train.train_iters=500000
+
+        # Multiple overrides for distributed training
+        python pretrain_kimi_k2.py \
+            model.tensor_model_parallel_size=2 \
+            model.pipeline_model_parallel_size=16 \
+            train.global_batch_size=4096
+    """
+    args, cli_overrides = parse_cli_args()
+
+    logger.info("Megatron-Bridge Kimi-K2 (1T) Pretraining Script with YAML & CLI Overrides")
+    logger.info("------------------------------------------------------------------")
+
+    # Load base configuration from the recipe as a Python dataclass
+    cfg: ConfigContainer = kimi_k2_pretrain_config(
+        optimizer_type=args.optimizer_type,
+        data_paths=args.data_paths,
+    ) if not args.proxy else kimi_k2_proxy_pretrain_config(
+        optimizer_type=args.optimizer_type,
+        data_paths=args.data_paths,
+    )
+    logger.info("Loaded base configuration")
+    cfg.to_yaml()
+
+    # Convert the initial Python dataclass to an OmegaConf DictConfig for merging
+    merged_omega_conf, excluded_fields = create_omegaconf_dict_config(cfg)
+
+    # Load and merge YAML overrides if a config file is provided
+    if args.config_file:
+        logger.debug(f"Loading YAML overrides from: {args.config_file}")
+        if not os.path.exists(args.config_file):
+            logger.error(f"Override YAML file not found: {args.config_file}")
+            sys.exit(1)
+        yaml_overrides_omega = OmegaConf.load(args.config_file)
+        merged_omega_conf = OmegaConf.merge(merged_omega_conf, yaml_overrides_omega)
+        logger.debug("YAML overrides merged successfully.")
+
+    # Apply command-line overrides using Hydra-style parsing
+    if cli_overrides:
+        logger.debug(f"Applying Hydra-style command-line overrides: {cli_overrides}")
+        merged_omega_conf = parse_hydra_overrides(merged_omega_conf, cli_overrides)
+        logger.debug("Hydra-style command-line overrides applied successfully.")
+
+    # Apply the final merged OmegaConf configuration back to the original ConfigContainer
+    logger.debug("Applying final merged configuration back to Python ConfigContainer...")
+    final_overrides_as_dict = OmegaConf.to_container(merged_omega_conf, resolve=True)
+    # Apply overrides while preserving excluded fields
+    apply_overrides(cfg, final_overrides_as_dict, excluded_fields)
+
+    # Display final configuration
+    logger.info("--- Final Merged Configuration ---")
+    cfg.to_yaml()
+    logger.info("----------------------------------")
+
```

Comment:
same here

we log the configuration after all overrides are applied in the training loop as well: https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/347624593b89e444930e411a88ca3a0f5dc99219/src/megatron/bridge/training/setup.py#L249-L253